### ISSUE TITLE
Block descriptions broken after SA-CORE-2019-011-013

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4071]
All module descriptions that use markup are broken. Some examples:

![Screen Shot 2019-09-20 at 3 26 33 pm](https://user-images.githubusercontent.com/2423362/65301493-21908f00-dbbb-11e9-8a0a-f0bacbaa98d1.png)
![Screen Shot 2019-09-20 at 3 26 13 pm](https://user-images.githubusercontent.com/2423362/65301494-21908f00-dbbb-11e9-96aa-ad708a00f284.png)
![Screen Shot 2019-09-20 at 3 26 00 pm](https://user-images.githubusercontent.com/2423362/65301495-22292580-dbbb-11e9-9d1a-ad69da6b934b.png)
![Screen Shot 2019-09-20 at 1 11 32 pm](https://user-images.githubusercontent.com/2423362/65296555-31eb3e80-dba8-11e9-91bc-ddf70ca70914.png)

The change that broke things was the addition of `check_plain()` when outputting block descriptions: https://github.com/backdrop/backdrop/commit/ed6bcc8cf56ed2f611d41f24522ce3a6b571c7a7#diff-e54edf78b887e6b7103a495d600bc139R1036

---
PR by @klonos: https://github.com/backdrop/backdrop/pull/2888

**Comment by klonos** [comment]
...turns out that all module descriptions that use markup are broken. Some more examples:

![Screen Shot 2019-09-20 at 3 26 33 pm](https://user-images.githubusercontent.com/2423362/65301493-21908f00-dbbb-11e9-8a0a-f0bacbaa98d1.png)
![Screen Shot 2019-09-20 at 3 26 13 pm](https://user-images.githubusercontent.com/2423362/65301494-21908f00-dbbb-11e9-96aa-ad708a00f284.png)
![Screen Shot 2019-09-20 at 3 26 00 pm](https://user-images.githubusercontent.com/2423362/65301495-22292580-dbbb-11e9-9d1a-ad69da6b934b.png)

The change that broke things was the addition of `check_plain()` when outputting block descriptions: https://github.com/backdrop/backdrop/commit/ed6bcc8cf56ed2f611d41f24522ce3a6b571c7a7#diff-e54edf78b887e6b7103a495d600bc139R1036

I have filed a PR that uses `filter_xss()` instead of `check_plain()`: https://github.com/backdrop/backdrop/pull/2888

This will need to be validated for security by a core committer.

**Comment by klonos** [comment]
I understand that this may be out of scope, but this is a single-line fix, so I thought that it'd be OK to include another one-liner UX fix in this PR (since it was what triggered the creation of this ticket): the link in the `Powered by Backdrop` block should have a `target="_blank"`, otherwise people clicking it may lose their block edits.

Please let me know if I should file a separate issue and PR for it, or if OK to include here.

**Comment by quicksketch** [comment]
Agreed, I ran into this today as well. filter_xss() is the right thing to be using in situations where some markup is allowed, like block descriptions.

However, I'm it seems like we're not treating block descriptions from custom blocks equally everywhere. If we're allowing markup in descriptions, we should show that markup on `admin/structure/block` as well. And we should update the help text/description on the description field when blocks are created.

**Comment by klonos** [comment]
> and seeing that we may need to do this in other places as well

We have such "sad" description fields all over the place. I will file a follow-up to get them done once this is merged.

**Comment by quicksketch** [comment]
PR looks good to me. I wouldn't mind a second set of eyes on it though.

**Comment by stpaultim** [comment]
I recreated this problem on a demo site and the tested the PR sandbox to see that the problems appears to be fixed. I viewed existing blocks with markup in the description and created my own reusable block with markup in the description on both sites. The PR seems to fix the problem it intends to fix. 

I don't feel qualified to comment on the actual changes to the code that accomplish this.

**Comment by klonos** [comment]
...moving the issue to the high priority column in the [Bugs project](https://github.com/backdrop/backdrop-issues/projects/16#card-27314299), since it's been RTBC'ed 😉

**Comment by quicksketch** [comment]
Merged into https://github.com/backdrop/backdrop/pull/2888 1.x and 1.14.x. Thanks @klonos and @stpaultim!
